src/main/java/com/modsen/software/ratings/service/RatingServiceImpl.java

```diff
@@ -74,16 +82,30 @@ public Rating update(Rating request) {
         if (!ratingRepository.existsById(request.getId()))
             throw new RatingNotExistsException(String.format("Rating with id = %d not exists", request.getId()));
 
+        if (!(driverServiceClient.getDriverById(request.getDriverId()).getStatusCode() == HttpStatus.OK
+                && passengerServiceClient.getPassengerById(request.getPassengerId()).getStatusCode() == HttpStatus.OK))
+            throw new RatingParticipantsNotExists("One or several participants not exist");
+
         request.setCreationDate(OffsetDateTime.now());
 
         return ratingRepository.save(request);
     }
 
     @Transactional
     public Rating changeRating(Long id, Byte rating) {
-        if (ratingRepository.changeRating(id, rating, new Date()) == 0)
+        if (ratingRepository.changeRating(id, rating, OffsetDateTime.now()) == 0)
             throw new RatingNotExistsException(String.format("Rating with id = %d not exists", id));
 
         return ratingRepository.findById(id).get();
     }
+
+    @Scheduled(fixedRate = 10000)
+    public void sendDriverRatingRecalculated() {
+        kafkaTemplate.send(KafkaTopics.RATING_TOPIC, new DriverRatingRecalculated(1L, 3.7f));
+    }
+
+    @Scheduled(fixedRate = 10000)
+    public void sendPassengerRatingRecalculated() {
+        kafkaTemplate.send(KafkaTopics.RATING_TOPIC, new PassengerRatingRecalculated(1L, 3.7f));
+    }
```

Comment:
Didn't get the logic. What is the purpose? Is it sending "mock" ratings using scheduler?

Reply:
At the moment, yes. In the future, I planned to do this: according to the planner, the user rating will be recalculated and the event about the new rating will be sent to the driver and passenger service.  After receiving the recalculated value, the services will update the rating information in the corresponding entity.